src/wallet/coinselection.h

```diff
@@ -145,6 +147,8 @@ struct CoinSelectionParams {
      * associated with the same address. This helps reduce privacy leaks resulting from address
      * reuse. Dust outputs are not eligible to be added to output groups and thus not considered. */
     bool m_avoid_partial_spends = false;
+    /** Filter transaction inputs by type */
+    std::unordered_set<OutputType> m_filter_inputs;
```

Comment:
if you do the filtering in `AvailableCoins`, we don't need to add this to `coinselection.h`